### PR TITLE
Feature/reset password first page

### PR DESCRIPTION
**Pull request by dmytro-chushko**
Reset Password First Page
endpoint forgotPassword in AuthModule
it validates the submitted email address, generates a token and sends an email

### CODE REVIEW

File: .development.env [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+MYSQL_PASSWORD=root
+MYSQL_PORT=3306
+MYSQL_ROOT_PASSWORD=root
+EMAIL_HOST=smtp.gmail.com
+EMAIL_PORT=587
+EMAIL_USER=dmytro.chushko@gmail.com
+EMAIL_PASS=fjpazpqhvkutoobz
+EMAIL_SENDER=dmytro.chushko@gmail.com
+EMAIL_SUBJECT=Reset password on Frelancer app
+RESET_PASSWOPRD_LINK=http://localhost:4200/
```
Comment (lyudad) [inline .development.env]: Снимок экрана 2022-11-16 в 22 38 32

File: src/main.ts [hunk 2]
```diff
@@ -5,7 +5,7 @@ import { SwaggerModule, DocumentBuilder } from '@nestjs/swagger';
 
 async function start() {
   const PORT = process.env.PORT || 8090;
-  const app = await NestFactory.create(AppModule);
+  const app = await NestFactory.create(AppModule, { cors: true });
```
Comment (lyudad) [inline src/main.ts]: Снимок экрана 2022-11-16 в 22 37 07

File: src/modules/auth/services/auth.service.ts [hunk 3]
```diff
@@ -55,4 +55,29 @@ export class AuthService implements IAuthService {
       throw new HttpException(`${error}`, HttpStatus.INTERNAL_SERVER_ERROR);
     }
   }
+
+  async generateToken(user: User) {
```
Comment (lyudad) [inline src/modules/auth/services/auth.service.ts]: Снимок экрана 2022-11-16 в 22 33 47

File: src/modules/mail/mail.module.ts [hunk 4]
```diff
@@ -20,7 +20,7 @@ import { ConfigModule } from '@nestjs/config';
       transport: {
         host: process.env.EMAIL_HOST,
         port: process.env.EMAIL_PORT,
-        secure: true,
+        // secure: true,
```
Comment (lyudad) [inline src/modules/mail/mail.module.ts]: Снимок экрана 2022-11-16 в 22 39 48

File: .development.env (hunk 1 above)
Comment (dmytro-chushko) [reply .development.env]: It is in .gitignor but it was in wrong place. I fixed it.

File: src/main.ts (hunk 2 above)
Comment (dmytro-chushko) [reply src/main.ts]: I didn't create this function. I only added {cors: true}. Didn't know I needed to modify the function, if it's not me created it. But think it's right. I will know for the future. I fixed it.

File: src/modules/auth/services/auth.service.ts (hunk 3 above)
Comment (dmytro-chushko) [reply src/modules/auth/services/auth.service.ts]: The same case. I just use this function, I didn't created it. But I will check in the future all the features that I have changed. I fixed it.

File: src/modules/mail/mail.module.ts (hunk 4 above)
Comment (dmytro-chushko) [reply src/modules/mail/mail.module.ts]: I missed it. I fixed it.

File: src/modules/auth/services/auth.service.ts [hunk 5]
```diff
@@ -55,4 +56,32 @@ export class AuthService implements IAuthService {
...
+  async forgotPassword(email: string): Promise<boolean> {
+    try {
+      const existingUser = await this.userService.findByEmail(email);
+      if (!existingUser) {
+        throw new BadRequestException(`user with email ${email} does not exists`);
+      }
+      const token = await this.generateToken(existingUser);
+      await this.mailService.sendActivationMail(
+        existingUser.email,
+        `${process.env.RESET_PASSWOPRD_LINK}resetpassword/${token.token}`,
```
Comment (lyudad) [inline src/modules/auth/services/auth.service.ts]: please check Nest documentation what is correct way to use env variables

**Comment by dmytro-chushko** [comment]
add  endpoint resetPassword
add service resetPassword

File: src/modules/auth/services/auth.service.ts (hunk 5 above)
Comment (dmytro-chushko) [reply src/modules/auth/services/auth.service.ts]: Fixed using env var
const resetPageLink = this.configService.get<string>('RESET_PASSWOPRD_LINK');

**Review by lyudad — APPROVED** [review]

**Pull request merged**